src/Reference.js

```diff
@@ -1,48 +1,70 @@
 import { getter } from 'property-expr';
 
-let validateName = d => {
-  if (typeof d !== 'string')
-    throw new TypeError("ref's must be strings, got: " + d);
+const prefixes = {
+  context: '$',
+  value: '.',
 };
 
 export default class Reference {
-  static isRef(value) {
-    return !!(value && (value.__isYupRef || value instanceof Reference));
-  }
+  constructor(key, options = {}) {
+    if (typeof key !== 'string')
+      throw new TypeError("ref's must be strings, got: " + key);
 
-  toString() {
-    return `Ref(${this.key})`;
+    this.key = key.trim();
+
+    this.isContext = this.key[0] === prefixes.context;
+    this.isValue = this.key[0] === prefixes.value;
+    this.isParent = this.key === '';
```

Comment:
I prefer to leave this out for now. This should be solved more robustly like #201

Reply:
Well, this is about treating all sources equally. If you can get value by `"."`, then path is empty string. If empty string works as path for value, why should it not work for context or parent? Why can't person just get parent or just get context object itself?

This is not about getting values from above a parent like `"../above"`.

Plus what is the status of mentioned PR, is this worked on in any way?

Reply:
> then path is empty string. 

That's only b/c you made `.` a prefix :P it wasn't meant as that, but as a single special value. I realize this covers, `'.foo'` as well but i'm not actually sure that is a good idea, it's fine if it falls out naturally tho i guess.

`''` wasn't ever an intentionally supported option, the idea is you can depend on siblings or context.

Reply:
> Why can i depend on sibling, but not on parent?

there isn't a reason generally, i think it's a good idea. I would like to scope that as a new feature, and ideally cover more cases instead of half implementing being able to reference parents.

> Why can't i get context as a whole (it might not even be an object)?

it has to be an object, i don't see any value in supporting other types.

Reply:
Mentioned PR is soon will be year old and as i understand its opener is not working on it right now.

`Lazy` does have access to `value`, `context` and `parent`.

Is there any problem with supporting getting a parent value with `""`  right now and then later introducing new stuff?

It might be a breaking change later, but really small and easy one to migrate.

If full feature support will be soon, then migration will be painless, if it will be in a distant future than having something working in meantime is worth it.

Am i missing some angle? Do you see any possible problems with migrations?

Reply:
I again, it's a distinct feature and needs to be evaluated on it's own. I don't really feel like adding new additions to the api that offer minimal value and would require a breaking change later as part of an unrelated PR. this isn't even a feature in Joi so i don't think it's a big deal to not support here at this moment.

Reply:
Ok. Removed ability to get a parent.

Reply:
But you still get context object by `"$"`. Is this ok or should be removed too?

Reply:
awesome thanks, I think that's fine for now, it's at least an older bug.